server/src-lib/Hasura/RQL/DDL/Metadata.hs

```diff
@@ -143,18 +144,20 @@ runClearMetadata _ = do
 
 data ReplaceMetadata
   = ReplaceMetadata
-  { aqTables         :: ![TableMeta]
-  , aqQueryTemplates :: ![DQ.CreateQueryTemplate]
-  , aqFunctions      :: !(Maybe [QualifiedFunction])
-  , aqRemoteSchemas  :: !(Maybe [TRS.AddRemoteSchemaQuery])
+  { aqTables           :: ![TableMeta]
+  , aqQueryTemplates   :: ![DQ.CreateQueryTemplate]
+  , aqFunctions        :: !(Maybe [QualifiedFunction])
+  , aqRemoteSchemas    :: !(Maybe [TRS.AddRemoteSchemaQuery])
+  , aqQueryCollections :: !(Maybe [DQC.CreateCollection])
+  , aqAllowlist        :: !(Maybe [DQC.CollectionName])
```

Comment:
To make it future proof, can we wrap this in a different type? So that the export metadata looks like this:
```yaml
allowlist:
- collection: coll1
- collection: coll2
```
So in future, we can add more options to allowlist api